ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp

```diff
@@ -49,20 +51,27 @@ MaxPool2D::MultiCore::cached_program_t max_pool_2d_multi_core_sharded_with_halo_
     tt::DataFormat out_df = datatype_to_dataformat_converter(output.get_dtype());
     uint32_t in_nbytes = datum_size(in_df);
     uint32_t out_nbytes = datum_size(out_df);
-    uint32_t in_nbytes_c = input_shape[3] * in_nbytes;                                      // row of input (channels)
-    uint32_t out_nbytes_c = output_shape[3] * out_nbytes;                                   // row of output (channels)
-    TT_ASSERT((in_nbytes_c & (in_nbytes_c - 1)) == 0, "in_nbytes_c should be power of 2");  // in_nbytes_c is power of 2
-    TT_ASSERT(
-        (out_nbytes_c & (out_nbytes_c - 1)) == 0, "out_nbytes_c should be power of 2");  // out_nbytes_c is power of 2
+    TensorMemoryLayout in_memory_layout = input.memory_config().memory_layout;
+
+    TT_FATAL((in_memory_layout == TensorMemoryLayout::HEIGHT_SHARDED) ||
```

Comment:
should check like this be in validate function?

Reply:
yeah all input tensor validation should go to validate function.

Reply:
sounds good will do